src/main.cpp

```diff
@@ -294,6 +297,97 @@ static void render_scene(Scene& scene, Vec2 const viewport_size)
   rendering::commit_draw();
 }
 
+void create_menu_window()
+{
+  ImGui::Begin("Window with menu bar and button toolbar", nullptr,
+               ImGuiWindowFlags_MenuBar);
+  if(ImGui::BeginMenuBar()) {
+    // Menu bar goes here...
+    ImGui::EndMenuBar();
+  }
+
+  // for each of the toolbar buttons
+  for(int i = 0; i < 7; ++i) {
```

Comment:
what if there are 8 kind?

Reply:
What are possible solutions? One would be defining X-macro with gate-types and creating names based on it, but this macro would have to be updated each time we add new gate type (now we need to update loop length and add new name for new gate) so it is somewhat comparable. 

Is X-macro the way or what do you suggest?

Reply:
Conditions met (to my knowledge at least)